z/z.go

```diff
@@ -18,22 +18,46 @@ package z
 
 import (
 	"context"
+	"log"
 	"sync"
 
 	"github.com/cespare/xxhash/v2"
 )
 
-// TODO: Figure out a way to re-use memhash for the second uint64 hash,
-// we already know that appending bytes isn't reliable for generating a
-// second hash (see Ristretto PR #88).
-// We also know that while the Go runtime has a runtime memhash128
-// function, it's not possible to use it to generate [2]uint64 or
-// anything resembling a 128bit hash, even though that's exactly what
-// we need in this situation.
-func KeyToHash(key interface{}) (uint64, uint64) {
-	if key == nil {
-		return 0, 0
+// GetKeyToHash will get the default KeyToHash function for the primitive types.
+func GetKeyToHash[K any](emptyKey K) func(K) (uint64, uint64) {
```

Comment:
I am wondering if it is possible to define this function at compile time somehow based on the type parameter. Is there a way we could achieve that using Go generics?

Reply:
i couldnt think of an elegant way since the KeyToHash function is expected to also be generic,
but let me think a little more

Reply:
@mangalaman93 there is no way i can find make this function generic sadley:(